tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -145,7 +145,45 @@ def get_resource_query(self):
             return {'filter': self.data.get('query')}
 
     def resources(self, query=None):
-        q = query or self.get_resource_query()
+        if hasattr(self.resource_type, 'parent'):
+            return self._fetch_resources_relationally(query)
```

Comment:
please don't call this fetch relationally, thats a very generic term where as this is primarily scoped to fetching parents.